### ISSUE TITLE
Return unique files and optionally copy them to all/selected directories that were searched.

### ISSUE THREAD

**Issue by pixelb** [pixelb/fslint#68]
Original [issue 69](https://code.google.com/p/fslint/issues/detail?id=69) created by pixelb on 2011-08-28T17:44:56.000Z:

Hi! Thank you for making FSlint! ^^
I have a feature request.

Can there be a way to list the files which are unique? Or also, &quot;almost unique&quot; files which exist in some, but not all, of the searched directories? Maybe there could be an option to copy the unique files to all/selected directories.

Not only could this be useful to find out what's not backed up, but also to figure out what has accidentally been deleted from or got corrupted on one/some of the locations.

I have probably everything stored on a few large hard drives as backup drives.
The problem for me is, I have 3+ drives which &quot;should&quot; have the exact same files on each, but which have all have been organized very differently, with most of all the files in different directories (and sometimes separate partitions) compared to the other drives.

I'd like to remove/hardlink duplicates with FSlint on each drive individually, and then search all of them together to make sure they all have at least one copy of each file. The normal Linux &quot;diff&quot; program doesn't help since it seems to expect the directories to be organized the same. But, I'm guessing FSlint would just have to return the results that did &quot;not&quot; make it past the checks for duplicates instead of the ones that did?

Maybe there could be a button to invert/toggle the results? If it takes too much memory for the program to store both lists for a toggle/inversion, then maybe it could be an option to specify before the search, like next to the &quot;recurse?&quot; checkbox.

This list would be great to know what I have backed up and what I don't.
For example, if I have 3 hard drives, with 2 of them each having one large partition, and the 3rd one split into 2 partitions, I could ask the program to search through those 4 base directories. If it finds that some files only exist in 1 place, maybe the program could have a button to copy the files to all (or just the selected) directories that were searched.  In my case, the directory structures are different and maybe it's easiest to either copy the files to the other directories, creating any needed parent directories, or maybe just a new user-defined directory in each.

Also, maybe it could also optionally return files which, for example, only exist in 1, 2, or 3 of the 4 searched directories. Not really always unique files then, but you'd know which backup drives/directories are missing something compared to the others.

So basically, a search or toggle/invert feature to show unique files and optionally copy them to the other searched areas (or optionally delete or save the list as usual, too). And, also an additional option to that feature to search for files which are in some, but not all of the searched directories (kind of &quot;almost&quot; unique)... That would be great for me.

Hope all that makes sense... I haven't been able to find a program with this feature yet, so I'm not sure how useful/practical it might be for others.

Thank you very much for your time and effort into this useful tool! Have a great day!!! ^_^
